confgenerator/config/config.go

```diff
@@ -265,6 +285,75 @@ func (c *configComponent) ValidateType(subagent string, component string, id str
 	return nil
 }
 
+func (r *LoggingReceiver) ValidateParameters(subagent string, component string, id string) error {
+	supportedParameters := supportedParameters[r.Type]
+	return validateParameters(*r, subagent, component, id, r.Type, supportedParameters)
+}
+
+func (p *LoggingProcessor) ValidateParameters(subagent string, component string, id string) error {
+	supportedParameters := supportedParameters[p.Type]
+	return validateParameters(*p, subagent, component, id, p.Type, supportedParameters)
+}
+
+func (r *MetricsReceiver) ValidateParameters(subagent string, component string, id string) error {
+	supportedParameters := supportedParameters[r.Type]
+	return validateParameters(*r, subagent, component, id, r.Type, supportedParameters)
+}
+
+type yamlField struct {
+	Name     string
+	Required bool
+	IsZero   bool
+}
+
+func nonZeroFields(sm reflect.Value) []yamlField {
+	var parameters []yamlField
+	tm := sm.Type()
+	if tm.NumField() != sm.NumField() {
+		panic(fmt.Sprintf("expected the number of fields in %v and %v to match", sm, tm))
+	}
+	for i := 0; i < tm.NumField(); i++ {
+		f := tm.Field(i)
+		t, _ := f.Tag.Lookup("yaml")
+		split := strings.Split(t, ",")
+		n := split[0]
+		annotations := split[1:]
+		if n == "-" {
+			continue
+		} else if n == "" {
+			n = strings.ToLower(f.Name)
+		}
+		v := sm.Field(i)
+		if sliceContains(annotations, "inline") {
+			// Expand inline structs.
+			parameters = append(parameters, nonZeroFields(v)...)
+		} else if f.Name[:1] != strings.ToLower(f.Name[:1]) { // skip private non-struct fields
```

Comment:
`if f.PkgPath == ""` is the idiomatic way to check if the field is exported (see the comment at https://pkg.go.dev/reflect#StructField)